REFERENCE.md

```diff
@@ -0,0 +1,363 @@
+sec.put("consumer_key", "yis0TYCu7U9V4o7M");
+sec.put("domain","localhost");
+
+reqData = new HashMap<String,String>();
+reqData.put("limit", "10");
+
+DataApi dataApi = new DataApi("https://data.learnosity.com/latest/itembank/items", sec, consumerSecret, reqData, "get");
+
+// you can also configure the Data API with a requestConfig as above
+RequestConfig requestConfig = RequestConfig.custom().setSocketTimeout(1000).build();
+
+dataApi.setRequestConfig(requestConfig);
+
+
+JSONObject response = dataApi.requestJSONObject();
+JSONObject res = new JSONObject(response.getString("body"));
+
+// You can also get a Remote object like this:
+Remote remote = dataApi.request();
+String body = remote.getBody();
+```
+
+## Troubleshooting
+
+We recommend that you use the Java JSON implementation which comes packaged in the sdk. In the past, we had several issues which were due to incompatible JSON implementations.
+
+
+## Creating jar files (using Eclipse Luna Release 4.4.0)
+
+The 'light' jar:
+
+In eclipse right click the java sdk project and select 'Export'. Under 'Java', select 'JAR file'. In the following screen only select the packages in the src (excluding the library folder) and click 'Finish'.
+
+The 'all' jar:
+
+In eclipse right click the java sdk project and select 'Export'. Under 'Java', select 'Runnable JAR file', click 'Next' and then 'Finish'. There are warnings that some libararies are included. This can be disregarded as the libraries are publicly available.
+
+## Building with Maven
+
+Run ```mvn package``` in your root directory (where pom.xml is located). This will put the light and all jar in the correct folders.
+
+## Using with a maven file based repository
+
+1.) Download the 'light' jar and the learnosity project pom file (pom.xml) into a temporary directory (say /home/temp)
+
+2.) Add a directory to your maven project to build the file-based repository into (e.g. ./learnosity-sdk-repo)
+
+3.) Install the learnosity SDK with maven into that directory by running the following (modify locations as appropriate):
+```bash
+mvn deploy:deploy-file -DgroupId=learnositysdk -DartifactId=learnositysdk -Dversion=0.16.3 -Durl=file:./learnosity-sdk-repo/ -DrepositoryId=learnosity-sdk-repo -DupdateReleaseInfo=true -Dfile=/home/temp/learnositysdk-0.16.3.jar -DpomFile=/home/temp/pom.xml
+```
+
+4.) Add the directory as a file based repository in your project pom.xml:
+```xml
+<repositories>
+    <repository>
+        <id>learnosity-sdk-repo</id>
+        <url>file://${project.basedir}/learnosity-sdk-repo</url>
+    </repository>
+</repositories>
+```
+
+5.) Don't forget to commit your file-based repository
+
+## Testing
+
+In order to check that you've added the required code correctly, you can download src/LearnositySdk/Test and try to run the test file. If it runs successfully, all should be fine.
+
+The Test.java class also gives some examples on how to use the sdk.
+
+## Further reading
```

Comment:
Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/Learnosity/learnosity-sdk-java/pullRequest?prid=8556170)
```suggestion
## Further reading

```